compiler/rustc_privacy/src/lib.rs

```diff
@@ -1504,7 +1504,7 @@ impl SearchInterfaceForPrivateItemsVisitor<'_> {
             };
             self.tcx.emit_spanned_lint(
                 lint,
-                hir_id,
+                self.tcx.hir().local_def_id_to_hir_id(self.item_def_id),
```

Comment:
Could you avoid calculating `hir_id` entirely?
This was its only uses besides the `parent_module(hir_id)` above which can be replaced with `parent_module_from_def_id(local_def_id)`.